### PR TITLE
{synapse} az synapse module migration to Microsoft Graph

### PR DESCRIPTION
**Pull request by kevinzz6**
**Related command**

az synapse 
**Description**

Migrating from Azure SDK's AAD Graph to Microsoft Graph
#22174 
**Testing Guide**

**History Notes**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
synapse

**Comment by kevinzz6** [comment]
I updated PR content and how to rerun checks?

File: src/azure-cli/azure/cli/command_modules/synapse/manual/operations/accesscontrol.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 from knack.util import CLIError
 from azure.cli.core.azclierror import InvalidArgumentValueError, ArgumentUsageError
 from azure.cli.core.util import is_guid
-from azure.graphrbac.models import GraphErrorException
+from azure.cli.command_modules.role._msgrpah import GraphError
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/synapse/manual/operations/accesscontrol.py:9]: `_msgrpah` is protected, so please don't import from it. 

Please see https://github.com/Azure/azure-cli/blob/dev/doc/microsoft_graph_client.md#error

Comment (kevinzz6) [reply src/azure-cli/azure/cli/command_modules/synapse/manual/operations/accesscontrol.py:9]: so I just need to remove this import statement and GraphError still is available?

Comment (kevinzz6) [reply src/azure-cli/azure/cli/command_modules/synapse/manual/operations/accesscontrol.py]: @jiasli please help to review my new commit. Thanks.

File: src/azure-cli/azure/cli/command_modules/synapse/manual/operations/accesscontrol.py [hunk 2]
```diff
@@ -141,20 +141,18 @@ def _resolve_object_id(cmd, assignee, fallback_to_object_id=False):
                            "Please using --assignee-object-id GUID to specify assignee accurately"
                            .format(assignee=assignee))
 
-        return result[0].object_id
-    except (CloudError, GraphErrorException):
+        return result[0]["id"]
+    except (CloudError, GraphError):
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/synapse/manual/operations/accesscontrol.py:145]: `CloudError` will never be raised by `GraphClient`. Capturing `GraphError` is enough.

File: src/azure-cli/azure/cli/command_modules/synapse/manual/operations/accesscontrol.py (hunk 1 above)
Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/synapse/manual/operations/accesscontrol.py]: > so I just need to remove this import statement and GraphError still is available?

Yes.

**Review by jiasli — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/synapse/manual/operations/accesscontrol.py [hunk 3]
```diff
@@ -141,20 +140,18 @@ def _resolve_object_id(cmd, assignee, fallback_to_object_id=False):
...
 
 
 def _get_object_stubs(graph_client, assignees):
-    from azure.graphrbac.models import GetObjectsParameters
     result = []
     assignees = list(assignees)  # callers could pass in a set
     for i in range(0, len(assignees), 1000):
-        params = GetObjectsParameters(include_directory_object_references=True, object_ids=assignees[i:i + 1000])
-        result += list(graph_client.objects.get_objects_by_object_ids(params))
+        result += list(graph_client.directory_object_get_by_ids(assignees[i:i + 1000]))
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/synapse/manual/operations/accesscontrol.py:154]: Has this code been tested?

This is not how `directory_object_get_by_ids` is used. See https://github.com/Azure/azure-cli/blob/33cbd01f75c7d4c3814de5a1cbf6f42de7890467/src/azure-cli/azure/cli/command_modules/role/custom.py#L1573-L1586

**Review by jiasli — CHANGES_REQUESTED** [review]
incorrect usage

**Review by jiasli — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**